File: java/src/main/java/com/pbdviewer/utils/pbclass/PbEnum.java
package com.pbdviewer.utils.pbclass;

import java.util.HashMap;
import java.util.Map;

public class PbEnum {

    private int index;
    private String name;
    private final Map<Integer, String> items = new HashMap<>();

    public int getIndex() {
        return index;
    }

    public void setIndex(int index) {
        this.index = index;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Map<Integer, String> getItems() {
        return items;
    }
}

File: html/js/pbclass/PbEnum.js
// PbEnum.js — port of Java PbEnum.java
export class PbEnum {
    constructor() {
        /** @type {number} */
        this.index = 0;
        /** @type {string} */
        this.name  = '';
        /** @type {Map<number, string>} */
        this.items = new Map();
    }
}
